circuit.py

```diff
@@ -4,24 +4,25 @@
 from fractions import Fraction
 from utils import decompose_pi_fraction, phase_frac_to_latex
 import pyzx as zx
+from transversal_op import *
 from typing import *
 
-class Circuit(object):
+class PauliCircuit(object):
     """
-    Class for representing quantum circuit.
+    Class for representing quantum Pauli circuit.
 
     """
 
     def __init__(self, no_of_qubit: int, name: str = '') -> None:
         """
-        Generating a circuit
+        Generating a Pauli circuit
 
         Args:
             no_of_qubit (int): Number of qubits in the circuit
             name (str, optional): Circuit's name (for display). Defaults to ''.
         """
         self.qubit_num:     int = no_of_qubit
-        self.ops:           List[PauliProductOperation] = list()
+        self.ops:           List[Union[Circuit, TransversalOp]] = list()
```

Comment:
In #93 we decided to go with:
![Screenshot_20210512_070859](https://user-images.githubusercontent.com/36427091/117989156-ef342800-b2f0-11eb-8400-d64b9496b097.png)
I thought that would have meant leaving the PauliCircuit class separate from transversal operations. So the mixing of transversal and non transversal would happen outside of the circuit class of which we would take segments :thinking: . Is there a particular reason why you went this way?